src/core/src/op/pad.cpp

```diff
@@ -102,7 +110,7 @@ void op::v1::Pad::validate_and_infer_types() {
     for (size_t i = 0; i < get_input_size(); i++)
         input_shapes.push_back(get_input_partial_shape(i));
     shape_infer(this, input_shapes, output_shapes);
-    set_output_type(0, get_input_element_type(0), output_shapes[0]);
+    set_output_type(0, result_et, output_shapes[0]);
```

Comment:
strange change in the core. I think we should not expect such changes in the core part

Reply:
That was found by tests and seems logical. For `Pad` in constant mode we may have value and the type of value should be the same as data type, so we can set such type as type of `Pad` output. And if we do reverse infer we can set this type to `Pad` input as well.